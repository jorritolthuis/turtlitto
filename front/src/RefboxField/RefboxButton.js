import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

/**
 * A button part of the refbox. Does a callback to its parent to handle onClick events
 * Author: S.A. Tanja
 * Author: G.W. van der Heijden
 *
 * Props:
 *  - teamColor: the team color of the button (cyan or magenta).
 *  - onClick: a function on what to do when the button is pressed
 *  - children: the children of the button
 */
const RefboxButton = props => {
  return (
    <Button
      className={props.className}
      teamColor={props.teamColor}
      onClick={() => {
        props.onClick();
      }}
    >
      {props.children}
    </Button>
  );
};

const Button = styled.button`
  background-color: ${props =>
    props.teamColor === "magenta"
      ? props.theme.refboxMagenta
      : props.theme.refboxCyan};
  justify-content: center;
  align-items: center;
  font-size: 4rem;
  padding: 1.5rem 0;
  border: 0.125rem solid;
  user-select: none;
`;

RefboxButton.propType = {
  teamColor: PropTypes.oneOf(["cyan", "magenta"]).isRequired,
  tag: PropTypes.oneOf([
    "KO",
    "FK",
    "GK",
    "TI",
    "C",
    "P",
    "Soft",
    "Medium",
    "Hard"
  ]).isRequired
};

export default RefboxButton;
