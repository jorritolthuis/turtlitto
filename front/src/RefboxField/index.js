import React from "react";
import Button from "./RefboxButton";
import styled, { css } from "styled-components";
import PropTypes from "prop-types";
import sendToServer from "../sendToServer";

const TAG_VALUES = {
  KO: "kick_off",
  FK: "free_kick",
  GK: "goal_kick",
  TI: "throw_in",
  C: "corner",
  P: "penalty",
  Soft: "soft",
  Medium: "medium",
  Hard: "hard"
};

/**
 * Constructs a refbox for a team consisting of 6 buttons:
 * - Kick off (KO)
 * - Free kick (FK)
 * - Goal kick (GK)
 * - Throw in (TI)
 * - Corner (C)
 * - Penalty (P)
 * Also takes care of handling commands that the buttons represents
 * Author: S.A. Tanja
 * Author: G.W. van der Heijden
 *
 * Props:
 *  - isPenalty: a boolean which indicates whether to go into penalty mode
 */
const RefboxField = props => {
  return (
    <Refboxes>
      <Refbox>
        {tags(props).map(tag => {
          return (
            <RefboxButton
              isPenalty={props.isPenalty}
              key={tag}
              teamColor={"magenta"}
              id={`${tag}_magenta`}
              onClick={() => {
                console.log(`${TAG_VALUES[tag]}_magenta`);
                sendToServer(`${TAG_VALUES[tag]}_magenta`, "command");
              }}
            >
              {tag}
            </RefboxButton>
          );
        })}
      </Refbox>
      <Refbox>
        {tags(props).map(tag => {
          return (
            <RefboxButton
              isPenalty={props.isPenalty}
              key={tag}
              teamColor={"cyan"}
              id={`${tag}_cyan`}
              onClick={() => {
                sendToServer(`${TAG_VALUES[tag]}_cyan`, "command");
              }}
            >
              {tag}
            </RefboxButton>
          );
        })}
      </Refbox>
    </Refboxes>
  );
};

const Refboxes = styled.div`
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  padding-top: 2rem;
`;

const Refbox = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  max-width: 12rem;

  /* TODO: define media query globally with sk */
  @media screen and (min-width: 360px) {
    max-width: 16rem;
  }
`;

const RefboxButton = styled(Button)`
  font-size: 2.5rem;

  /* TODO: define media query globally with sk */
  @media screen and (min-width: 360px) {
    font-size: 4rem;
  }

  ${props =>
    props.isPenalty
      ? css`
          flex-basis: 100%;
        `
      : css`
          flex-basis: 50%;
          max-width: 6rem;
          max-height: 6rem;

          /* TODO: define media query globally with sk */
          @media screen and (min-width: 360px) {
            max-width: 8rem;
            max-height: 8rem;
            font-size: 4rem;
          }
        `};
`;

const tags = props => {
  if (props.isPenalty) {
    return ["Soft", "Medium", "Hard"];
  } else {
    return ["KO", "FK", "GK", "TI", "C", "P"];
  }
};

RefboxField.propType = {
  isPenalty: PropTypes.bool.isRequired
};

export default RefboxField;
