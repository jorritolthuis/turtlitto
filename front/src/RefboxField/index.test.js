import React from "react";
import RefboxField from ".";
import renderer from "react-test-renderer";

describe("RefboxField", () => {
  it("shows a cyan refbox field with all buttons", () => {
    const component = renderer.create(
      <RefboxField
        teamColor="cyan"
        onClick={(tag, color) => {}}
        isPenalty={false}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("shows a magenta refbox field with all buttons", () => {
    const component = renderer.create(
      <RefboxField
        teamColor="magenta"
        onClick={(tag, color) => {}}
        isPenalty={false}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("shows a magenta refbox with penalty buttons", () => {
    const component = renderer.create(
      <RefboxField
        teamColor="magenta"
        onClick={(tag, color) => {}}
        isPenalty={true}
      />
    );
  });
});
