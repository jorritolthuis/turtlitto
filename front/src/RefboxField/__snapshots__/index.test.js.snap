// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefboxField shows a cyan and magenta refbox field with all buttons 1`] = `
.c3 {
  background-color: #ff00ff;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  padding: 1.5rem 0;
  border: 0.125rem solid #c800c8;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:active {
  background-color: #c800c8;
}

.c4 {
  background-color: #00ffff;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  padding: 1.5rem 0;
  border: 0.125rem solid #00c8c8;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c4:active {
  background-color: #00c8c8;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  padding-top: 2rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 12rem;
}

.c2 {
  font-size: 2.5rem;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  max-width: 6rem;
  max-height: 6rem;
}

@media (min-width:576px) {
  .c0 {
    padding-top: 0;
  }
}

@media (min-width:376px) {
  .c1 {
    max-width: 16rem;
  }
}

@media (min-width:376px) {
  .c2 {
    font-size: 4rem;
  }
}

@media (min-width:376px) {
  .c2 {
    max-width: 8rem;
    max-height: 8rem;
    font-size: 4rem;
  }
}

<RefboxField
  isPenalty={false}
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <Styled(RefboxButton)
            id="KO_magenta"
            isPenalty={false}
            key="KO"
            onClick={[Function]}
            teamColor="magenta"
          >
            <RefboxButton
              className="c2"
              id="KO_magenta"
              isPenalty={false}
              onClick={[Function]}
              teamColor="magenta"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="magenta"
              >
                <button
                  className="c2 c3"
                  onClick={[Function]}
                >
                  KO
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="FK_magenta"
            isPenalty={false}
            key="FK"
            onClick={[Function]}
            teamColor="magenta"
          >
            <RefboxButton
              className="c2"
              id="FK_magenta"
              isPenalty={false}
              onClick={[Function]}
              teamColor="magenta"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="magenta"
              >
                <button
                  className="c2 c3"
                  onClick={[Function]}
                >
                  FK
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="GK_magenta"
            isPenalty={false}
            key="GK"
            onClick={[Function]}
            teamColor="magenta"
          >
            <RefboxButton
              className="c2"
              id="GK_magenta"
              isPenalty={false}
              onClick={[Function]}
              teamColor="magenta"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="magenta"
              >
                <button
                  className="c2 c3"
                  onClick={[Function]}
                >
                  GK
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="TI_magenta"
            isPenalty={false}
            key="TI"
            onClick={[Function]}
            teamColor="magenta"
          >
            <RefboxButton
              className="c2"
              id="TI_magenta"
              isPenalty={false}
              onClick={[Function]}
              teamColor="magenta"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="magenta"
              >
                <button
                  className="c2 c3"
                  onClick={[Function]}
                >
                  TI
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="C_magenta"
            isPenalty={false}
            key="C"
            onClick={[Function]}
            teamColor="magenta"
          >
            <RefboxButton
              className="c2"
              id="C_magenta"
              isPenalty={false}
              onClick={[Function]}
              teamColor="magenta"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="magenta"
              >
                <button
                  className="c2 c3"
                  onClick={[Function]}
                >
                  C
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="P_magenta"
            isPenalty={false}
            key="P"
            onClick={[Function]}
            teamColor="magenta"
          >
            <RefboxButton
              className="c2"
              id="P_magenta"
              isPenalty={false}
              onClick={[Function]}
              teamColor="magenta"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="magenta"
              >
                <button
                  className="c2 c3"
                  onClick={[Function]}
                >
                  P
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="c1"
        >
          <Styled(RefboxButton)
            id="KO_cyan"
            isPenalty={false}
            key="KO"
            onClick={[Function]}
            teamColor="cyan"
          >
            <RefboxButton
              className="c2"
              id="KO_cyan"
              isPenalty={false}
              onClick={[Function]}
              teamColor="cyan"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="cyan"
              >
                <button
                  className="c2 c4"
                  onClick={[Function]}
                >
                  KO
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="FK_cyan"
            isPenalty={false}
            key="FK"
            onClick={[Function]}
            teamColor="cyan"
          >
            <RefboxButton
              className="c2"
              id="FK_cyan"
              isPenalty={false}
              onClick={[Function]}
              teamColor="cyan"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="cyan"
              >
                <button
                  className="c2 c4"
                  onClick={[Function]}
                >
                  FK
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="GK_cyan"
            isPenalty={false}
            key="GK"
            onClick={[Function]}
            teamColor="cyan"
          >
            <RefboxButton
              className="c2"
              id="GK_cyan"
              isPenalty={false}
              onClick={[Function]}
              teamColor="cyan"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="cyan"
              >
                <button
                  className="c2 c4"
                  onClick={[Function]}
                >
                  GK
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="TI_cyan"
            isPenalty={false}
            key="TI"
            onClick={[Function]}
            teamColor="cyan"
          >
            <RefboxButton
              className="c2"
              id="TI_cyan"
              isPenalty={false}
              onClick={[Function]}
              teamColor="cyan"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="cyan"
              >
                <button
                  className="c2 c4"
                  onClick={[Function]}
                >
                  TI
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="C_cyan"
            isPenalty={false}
            key="C"
            onClick={[Function]}
            teamColor="cyan"
          >
            <RefboxButton
              className="c2"
              id="C_cyan"
              isPenalty={false}
              onClick={[Function]}
              teamColor="cyan"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="cyan"
              >
                <button
                  className="c2 c4"
                  onClick={[Function]}
                >
                  C
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="P_cyan"
            isPenalty={false}
            key="P"
            onClick={[Function]}
            teamColor="cyan"
          >
            <RefboxButton
              className="c2"
              id="P_cyan"
              isPenalty={false}
              onClick={[Function]}
              teamColor="cyan"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="cyan"
              >
                <button
                  className="c2 c4"
                  onClick={[Function]}
                >
                  P
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
        </div>
      </styled.div>
    </div>
  </styled.div>
</RefboxField>
`;

exports[`RefboxField shows a refbox with penalty buttons 1`] = `
.c3 {
  background-color: #ff00ff;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  padding: 1.5rem 0;
  border: 0.125rem solid #c800c8;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c3:active {
  background-color: #c800c8;
}

.c4 {
  background-color: #00ffff;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  padding: 1.5rem 0;
  border: 0.125rem solid #00c8c8;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c4:active {
  background-color: #00c8c8;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  padding-top: 2rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 12rem;
}

.c2 {
  font-size: 2.5rem;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
}

@media (min-width:576px) {
  .c0 {
    padding-top: 0;
  }
}

@media (min-width:376px) {
  .c1 {
    max-width: 16rem;
  }
}

@media (min-width:376px) {
  .c2 {
    font-size: 4rem;
  }
}

<RefboxField
  isPenalty={true}
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <Styled(RefboxButton)
            id="Soft_magenta"
            isPenalty={true}
            key="Soft"
            onClick={[Function]}
            teamColor="magenta"
          >
            <RefboxButton
              className="c2"
              id="Soft_magenta"
              isPenalty={true}
              onClick={[Function]}
              teamColor="magenta"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="magenta"
              >
                <button
                  className="c2 c3"
                  onClick={[Function]}
                >
                  Soft
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="Medium_magenta"
            isPenalty={true}
            key="Medium"
            onClick={[Function]}
            teamColor="magenta"
          >
            <RefboxButton
              className="c2"
              id="Medium_magenta"
              isPenalty={true}
              onClick={[Function]}
              teamColor="magenta"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="magenta"
              >
                <button
                  className="c2 c3"
                  onClick={[Function]}
                >
                  Medium
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="Hard_magenta"
            isPenalty={true}
            key="Hard"
            onClick={[Function]}
            teamColor="magenta"
          >
            <RefboxButton
              className="c2"
              id="Hard_magenta"
              isPenalty={true}
              onClick={[Function]}
              teamColor="magenta"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="magenta"
              >
                <button
                  className="c2 c3"
                  onClick={[Function]}
                >
                  Hard
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="c1"
        >
          <Styled(RefboxButton)
            id="Soft_cyan"
            isPenalty={true}
            key="Soft"
            onClick={[Function]}
            teamColor="cyan"
          >
            <RefboxButton
              className="c2"
              id="Soft_cyan"
              isPenalty={true}
              onClick={[Function]}
              teamColor="cyan"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="cyan"
              >
                <button
                  className="c2 c4"
                  onClick={[Function]}
                >
                  Soft
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="Medium_cyan"
            isPenalty={true}
            key="Medium"
            onClick={[Function]}
            teamColor="cyan"
          >
            <RefboxButton
              className="c2"
              id="Medium_cyan"
              isPenalty={true}
              onClick={[Function]}
              teamColor="cyan"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="cyan"
              >
                <button
                  className="c2 c4"
                  onClick={[Function]}
                >
                  Medium
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
          <Styled(RefboxButton)
            id="Hard_cyan"
            isPenalty={true}
            key="Hard"
            onClick={[Function]}
            teamColor="cyan"
          >
            <RefboxButton
              className="c2"
              id="Hard_cyan"
              isPenalty={true}
              onClick={[Function]}
              teamColor="cyan"
            >
              <styled.button
                className="c2"
                onClick={[Function]}
                teamColor="cyan"
              >
                <button
                  className="c2 c4"
                  onClick={[Function]}
                >
                  Hard
                </button>
              </styled.button>
            </RefboxButton>
          </Styled(RefboxButton)>
        </div>
      </styled.div>
    </div>
  </styled.div>
</RefboxField>
`;

exports[`When clicked, a cyan C button should pass \`corner_cyan\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"corner_cyan\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;

exports[`When clicked, a cyan FK button should pass \`free_kick_cyan\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"free_kick_cyan\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;

exports[`When clicked, a cyan GK button should pass \`goal_kick_cyan\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"goal_kick_cyan\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;

exports[`When clicked, a cyan KO button should pass \`kick_off_cyan\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"kick_off_cyan\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;

exports[`When clicked, a cyan P button should pass \`penalty_cyan\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"penalty_cyan\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;

exports[`When clicked, a cyan TI button should pass \`throw_in_cyan\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"throw_in_cyan\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;

exports[`When clicked, a magenta C button should pass \`corner_magenta\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"corner_magenta\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;

exports[`When clicked, a magenta FK button should pass \`free_kick_magenta\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"free_kick_magenta\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;

exports[`When clicked, a magenta GK button should pass \`goal_kick_magenta\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"goal_kick_magenta\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;

exports[`When clicked, a magenta KO button should pass \`kick_off_magenta\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"kick_off_magenta\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;

exports[`When clicked, a magenta P button should pass \`penalty_magenta\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"penalty_magenta\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;

exports[`When clicked, a magenta TI button should pass \`throw_in_magenta\` to the 'sendToServer' function 1`] = `
Object {
  "body": "\\"throw_in_magenta\\"",
  "headers": Headers {
    "map": Object {
      "authorization": "Basic dXNlcjp1bmRlZmluZWQ=",
    },
  },
  "method": "POST",
}
`;
